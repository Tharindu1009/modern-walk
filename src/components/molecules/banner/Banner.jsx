import './styles/banner.scss';
import { useState } from 'react';
import Label from '../../atoms/label/Label';
import CN from "classnames";
import { labelTypes, productCategoryTypes } from '../../../constants';

function Banner({type, text}) {
    const [categoryBannerLabel] = useState(labelTypes.categoryBanner);

    // Appling relevant style class by checking the "type" prop
    const BannerClasses = CN({
        "banner banner__teal": type === productCategoryTypes.mensClothing,
        "banner banner__pink": type === productCategoryTypes.womensClothing,
    });

    return (
        <div data-testid="banner" className={BannerClasses}>
            <Label type={categoryBannerLabel} text={text} />
        </div>
    )
}

export default Banner;