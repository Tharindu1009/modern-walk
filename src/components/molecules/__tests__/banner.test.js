import { render, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import Banner from "../banner/Banner";

describe('Banner', () => {
    afterEach(() => {
        cleanup();
    })

    it('should render mens clothing banner component correctly', () => {
        const { getByTestId } = render(<Banner type="men's clothing" text="Men's Clothing" />);
        const bannerElement = getByTestId('banner');
        expect(bannerElement).toBeInTheDocument();
    });

    it('should render womens clothing banner component correctly', () => {
        const { getByTestId } = render(<Banner type="women's clothing" text="Women's Clothing" />);
        const bannerElement = getByTestId('banner');
        expect(bannerElement).toBeInTheDocument();
    });

    it('matches banner snapshot', () => {
        const tree = TestRenderer.create(<Banner type="men's clothing" text="Men's Clothing" />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});