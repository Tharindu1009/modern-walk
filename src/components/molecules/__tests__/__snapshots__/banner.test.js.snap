// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner matches banner snapshot 1`] = `
<div
  className="banner banner__teal"
  data-testid="banner"
>
  <div
    className=""
    data-testid="label"
  >
     
    Men's Clothing
  </div>
</div>
`;
